### PR TITLE
Allow passing in display_names for a file

### LINKED ISSUE #3: Modify UploadFile
Hello
I perfer to using `filename_fixed = file.replace(" ", "_")` instead of `filename_fixed = os.path.basename(file).replace(" ", "_")`.

In my case, I have many folders in submission folder, but there is file which have same name`(MainActivity.java)` in different folder. So the result report can not tell me which folder it is in.More importantly, in report, the different link `MainActivity.java` point to the same `MainActivity.java`!

Directory Structure Example
```
                      - No.1 - MainActivity.java
                      |
Submission          - - No.2 - MainActivity.java
                      |
                     -  No.3- MainActivity.java
```

If you agree, I would create my pull request :)

### PR DESCRIPTION
**Pull request by nareddyt**
As discussed in issue #3 

Closes #3

### CODE REVIEW

File: mosspy/moss.py [hunk 1]
```diff
@@ -69,40 +69,46 @@ def setDirectoryMode(self, mode):
...
+            self.files.append((file, None))
 
     def getLanguages(self):
-        return languages
+        return self.languages
+
+    def uploadFile(self, s, file_path, display_name, file_id):
+        if display_name is None:
+            # If no display name added by user, default to file path
+            display_name = os.path.basename(file_path).replace(" ", "_")
```
Comment (soachishti) [inline mosspy/moss.py]: @nareddyt I tested code against the following directory hierarchy and got into same issue stated in #3.  
```
submission\
submission\p146011\a01.py
submission\p146099\a01.py
```
I think, this `basename` needs to be removed? What do you say...

Comment (nareddyt) [reply mosspy/moss.py]: Ahh yes, let me change that

**Comment by nareddyt** [comment]
How about now? @soachishti @YsingYang

**Comment by YsingYang** [comment]
```
    def getLanguages(self):
        return languages
```

I have a question, This function is used to return languages ? So, Does it should be self.languages?

**Comment by nareddyt** [comment]
I don't believe that ever worked. Without the `self` keyword, python just looks for a variable defined in the function itself. However, `languages` is defined in the class scope.

Notice that everywhere else `languages` is used, you did write `self.languages`

Would you like me to test it? Do you want me to make a separate pull request for this?

**Comment by YsingYang** [comment]
My fault, I have clone a older version...So, I don't notice you have fixed it...

**Pull request merged**